src/main/scala/gov/wicourts/json/formlet/Forms.scala

```diff
@@ -127,105 +125,115 @@ object Forms {
 
   def listM[M[_], A](
     template: ObjectFormlet[M, A],
-    defaultValue: List[ObjectFormlet[M, A]]
+    defaultValue: List[ObjectFormlet[M, A]],
   )(
-    implicit M: Applicative[M]
+    implicit M: Applicative[M],
   ): JsonFormlet[M, JsonArrayBuilder, ValidationErrors, List[A]] =
     Formlet { c =>
       val l: List[(Option[Cursor], ObjectFormlet[M, A])] =
         if (c.isEmpty || c.map(_.focus).exists(_.string.exists(_ === "")))
           defaultValue.map((c, _))
         else {
           type X = Vector[(Option[Cursor], ObjectFormlet[M, A])]
-          c
-            .flatMap(_.downArray)
+          c.flatMap(_.downArray)
             .map(c =>
               traverseBreak(
                 c,
                 Kleisli[State[X, ?], Cursor, Option[Cursor]](c =>
-                  State(l => (l :+ ((c.some, template)), c.right))
-                )
-              ).apply(Vector()).toList
+                  State(l => (l :+ ((c.some, template)), c.some)),
+                ),
+              ).apply(Vector()).toList,
             )
             .getOrElse(Nil)
         }
 
-      val X = M
-        .compose[Tuple2[JsonArrayBuilder, ?]]
-        .compose[Validation[ValidationErrors, ?]]
-      M.map(X.traverse(l.zipWithIndex) { case ((i, x), idx) =>
-        M.map(x.mapResult((r, v) => (
-          r.leftMap(o => ValidationErrors.arrayErrors(List((idx, o)))),
-          JsonArrayBuilder.item(v.toJson)
-        )).run(i))(_.swap)
-      })(_.swap)
+      type G[AA] = M[(JsonArrayBuilder, Validated[ValidationErrors, AA])]
+      implicit val applicativeG =
```

Comment:
Can we just stick with `X` (and no implicit) here?

Reply:
I haven't found a way to get rid of G, but I can stick with X as the name to make the diff a little smaller.

Reply:
Looks good.  (Am I missing a way to mark a MR comment as resolved?)

Reply:
Not sure. I have a resolve button below each thread.